cads_adaptors/adaptors/cams_solar_rad/functions.py

```diff
@@ -0,0 +1,483 @@
+import hashlib
+import logging
+import re
+import time
+import traceback
+
+import jinja2
+
+
+class BadRequest(Exception):
```

Comment:
Would be good to move to the excpetions.py so that we have a single source of the exceptions raised from cads-adaptors. These are used for organising the splunk messages, so it's also good to make them Adaptor specific as I did with the others

Reply:
The BadRequest and NoData exceptions are caught before getting outside the adaptor.